components/ai_chat/resources/page/components/open_external_link_modal/index.tsx

```diff
@@ -8,76 +8,48 @@ import Button from '@brave/leo/react/button'
 import Checkbox from '@brave/leo/react/checkbox'
 import Dialog from '@brave/leo/react/dialog'
 import { getLocale } from '$web-common/locale'
+import { AIChatContext } from '../../state/ai_chat_context'
 import styles from './style.module.scss'
-import getAPI from '../../api'
 import {
   IGNORE_EXTERNAL_LINK_WARNING_KEY //
 } from '../../../common/constants'
+import { ConversationContext } from '../../state/conversation_context'
 
-export default function OpenExternalLinkModal() {
-  // API
-  const api = getAPI()
+type Props = Pick<
+  ConversationContext,
+  'generatedUrlToBeOpened' | 'setGeneratedUrlToBeOpened'
+> &
+  Pick<AIChatContext, 'uiHandler'>
+
+interface OpenExternalLinkModalProps {
+  context: Props
+}
+
+export default function OpenExternalLinkModal(
+  props: OpenExternalLinkModalProps
+) {
+  const { context } = props
 
   // State
   const [ignoreChecked, setIgnoreChecked] = React.useState(false)
-  const [openingExternalLinkURL, setOpeningExternalLinkURL] = React.useState('')
-
-  // Local storage
-  const ignoreExternalLinkWarning = JSON.parse(
-    localStorage.getItem(IGNORE_EXTERNAL_LINK_WARNING_KEY) ?? 'false'
-  )
 
   // Methods
-  const handleOpenExternalLink = (url: string) => {
-    if (chrome.tabs !== undefined) {
-      chrome.tabs.create({ url }, () => {
-        if (chrome.runtime.lastError) {
-          console.error(
-            'tabs.create failed: ' + chrome.runtime.lastError.message
-          )
-        }
-      })
-    } else {
-      // Tabs.create is desktop specific. Using window.open for android.
-      window.open(url, '_blank', 'noopener noreferrer')
-    }
-  }
 
   const onOpenClicked = React.useCallback(() => {
     if (ignoreChecked) {
       localStorage.setItem(IGNORE_EXTERNAL_LINK_WARNING_KEY, 'true')
     }
```

Comment:
if we've moved the `localStorage.getItem(IGNORE_EXTERNAL_LINK_WARNING_KEY)` to the context, should we also move the `localStorage.setItem(IGNORE_EXTERNAL_LINK_WARNING_KEY)` so the management of that is in one module and not split between ui components and that?

Reply:
Moved to `conversationContext`.